bin/normaliseArticles.js

```diff
@@ -39,27 +38,11 @@ This is a stub. [Help our community expand it](https://github.com/freecodecamp/g
 /* eslint-enable max-len */
 
 function normaliseLinks(content) {
-  let anchored = content.slice(0);
-  const links = content.match(markdownLinkRE);
-
-  if (links) {
-  links
-    .filter(x => !x.startsWith('!'))
-    .filter(x => x.match(httpsRE))
-    .map(str => {
-      // raw will look like: 
-      // [ '[guides website', 'https://guide.freecodecamp.org)' ]
-      const raw = str.slice(0).split('](');
-      const formatted = [ raw[0].replace('[', ''), raw[1].replace(')', '') ];
-      const [ childText, url ] = formatted;
-      const anchor = (
-        `<a href='${url}' target='_blank' rel='nofollow'>${childText}</a>`
-      );
-      anchored = anchored.replace(str, anchor);
-    });
-  }
-
-  return anchored;
+  return content.replace(markdownLinkRE, (match, before, text, url) => {
+    return markdownLinkRE.test(text) ? match : (
```

Comment:
I don't understand this bit?

```
if (there is a markdown link in the child text) {
  don't replace anything
} else {
  format the link
}
```

Have I got that right?

Reply:
Yes, correct! It's a bit of a hack to try to help with the issue of having multiple links on one line.

Updating the regex to not allow `)` in the link destination would break for some links: https://en.wikipedia.org/wiki/Java_(programming_language)

If there are two links on a single line, this will still fail:
```
[link 1](http://example.com/1) text [link 2](http://example.com/2) text
```
`text` will equal `link 1](http://example.com/1) text [link 2`. I can't think of a way to probably catch this using regular expressions. Interestingly, I seems like this problem hasn't shown itself in the real world yet.

In the event that there are three or more links on one line (only [this closure article](https://github.com/freeCodeCamp/guides/blob/a36162f4ec524cc9ec7d20b1e11d4c3c2e4e276f/src/pages/clojure/lists-are-everything/index.md#when-to-use-a-list) has that), this approach will make it have the same result as without this change: the Markdown links are left untouched. The `regex.test` returns true, because the full text of the middle link is in the `text`.
The ternary operator is only for this case, three or more links on one line.

I'm sorry, I should've explained that more clearly in the original comment.